# Project Submission

**Please use this notebook for your submission.**

Make sure to fill out all the required fields and to answer all the questions.

You will be working with a dataset that includes the results of international football matches from 1872 to 2024. The matches range from FIFA World Cups to regular friendly matches. 

At the end of this project, you will have answered the following questions:

1. What is the average number of goals for the home team? 
2. What is the average number of goals for the away team? 
3. What is the country with the highest overall home score?
4. What are the top 3 type of tournaments?
5. Which country has the highest overall FIFA World Cup goals?


**Submission Requirements:**

- Make sure that you run all cells with code in your notebook before submitting.
- You can add additional cells of code if you want, but make sure to clean up your notebook, and only leave the code required to answer the questions.


_Original Data Source: [Kaggle](https://www.kaggle.com/datasets/martj42/international-football-results-from-1872-to-2017/data)_

## Step 1: Getting Started

You will mainly work with a Python library called Pandas. Pandas is a powerful library that allows us to manipulate data. In order to use Pandas, you first have to import it. 

In [95]:
import pandas as pd

## Step 2: Loading and Exporing the Data

You can use Pandas to explore and manipulate the _results.csv_ file. 
You first have to load the csv file into a Pandas dataframe, so you can then analyize the data. 

When using `pd.read_csv()`, make sure you inlcude the correct path to the csv file, depending on where you saved it when you downloaded it.

In [98]:
# First: Load the csv file into a Pandas dataframe (df)
df = pd.read_csv('results.csv')

In [100]:
# Explore the df

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45315 entries, 0 to 45314
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   date        45315 non-null  object
 1   home_team   45315 non-null  object
 2   away_team   45315 non-null  object
 3   home_score  45315 non-null  int64 
 4   away_score  45315 non-null  int64 
 5   tournament  45315 non-null  object
 6   city        45315 non-null  object
 7   country     45315 non-null  object
 8   neutral     45315 non-null  bool  
dtypes: bool(1), int64(2), object(6)
memory usage: 2.8+ MB


## Step 3: Data Analysis

### Q1: What is the average number of goals for the home team?

Hint: You can call the `mean()` method on the _'home_score'_ column.

In [105]:
df['home_score'].mean()



1.7393136930376256

the mean of the home score is 1.7393136930376256

### Q2: What is the average number of goals for the away team?

Hint: You can call the `mean()` method on the _'away_score'_ column.

In [109]:
df['away_score'].mean()



1.1782412004854905

the mean of the away score is 1.1782412004854905

### Q3: What is the country with the highest overall home score?

Hint: You will need to group the data by country, then get the sum of home scores for each country. You can use the `idxmax()` function to find the index(country) with the highest sum. 

In [113]:
df.groupby('country')['home_score'].idxmax().head(1)

country
Afghanistan    35813
Name: home_score, dtype: int64

Afghanistan

### Q4: What are the top 3 type of tournaments?

Hint: You can use the `value_counts()` method to count the occurences of each unique value in the _'tournament'_ column.

In [117]:
df['tournament'].value_counts().head(3)



tournament
Friendly                        17773
FIFA World Cup qualification     8016
UEFA Euro qualification          2815
Name: count, dtype: int64

1.Friendly 2.FIFA WC qualifications 3. UEFA Euro qualifications

### Q5: Which country has the highest overall FIFA World Cup goals?

To answer this question, consider breaking down your solution into 3 steps.

**5.1 Create a new dataframe (fifa_df) that only contains rows where the "tournament" column == 'FIFA World Cup'.**

In [193]:
fifa_df = df[['tournament','country','home_team','away_team','home_score','away_score']].copy()

In [198]:
fifa_df=fifa_df.query('tournament =="FIFA World Cup"')
print(fifa_df)

           tournament  country  home_team      away_team  home_score  \
1311   FIFA World Cup  Uruguay    Belgium  United States           0   
1312   FIFA World Cup  Uruguay     France         Mexico           4   
1313   FIFA World Cup  Uruguay     Brazil     Yugoslavia           1   
1314   FIFA World Cup  Uruguay       Peru        Romania           1   
1315   FIFA World Cup  Uruguay  Argentina         France           1   
...               ...      ...        ...            ...         ...   
44349  FIFA World Cup    Qatar    England         France           1   
44351  FIFA World Cup    Qatar  Argentina        Croatia           3   
44352  FIFA World Cup    Qatar     France        Morocco           2   
44356  FIFA World Cup    Qatar    Croatia        Morocco           2   
44358  FIFA World Cup    Qatar  Argentina         France           3   

       away_score  
1311            3  
1312            1  
1313            2  
1314            3  
1315            0  
...           .

**5.2 In your new fifa_df, create a new _'total_score'_ column that sums _'home_score'_ + _'away_score'_ for each row.**

PS. If you get a "SettingWithCopyWarning", please ignore it.

In [201]:
fifa_df['total_score']=fifa_df.apply(lambda row: row.home_score + row.away_score, axis=1)
print(fifa_df)

           tournament  country  home_team      away_team  home_score  \
1311   FIFA World Cup  Uruguay    Belgium  United States           0   
1312   FIFA World Cup  Uruguay     France         Mexico           4   
1313   FIFA World Cup  Uruguay     Brazil     Yugoslavia           1   
1314   FIFA World Cup  Uruguay       Peru        Romania           1   
1315   FIFA World Cup  Uruguay  Argentina         France           1   
...               ...      ...        ...            ...         ...   
44349  FIFA World Cup    Qatar    England         France           1   
44351  FIFA World Cup    Qatar  Argentina        Croatia           3   
44352  FIFA World Cup    Qatar     France        Morocco           2   
44356  FIFA World Cup    Qatar    Croatia        Morocco           2   
44358  FIFA World Cup    Qatar  Argentina         France           3   

       away_score  total_score  
1311            3            3  
1312            1            5  
1313            2            3  
131

**5.3 Group your data by country, then get the sum of the _'total_score'_ column. From there, you can use `idxmax()` to find the country with the highest _'total_score'_.**

In [204]:
# Code here
fifa_df.groupby('country')['total_score'].idxmax().head(1)


country
Argentina    10756
Name: total_score, dtype: int64